mobile/status_request_log.go

```diff
@@ -0,0 +1,51 @@
+package statusgo
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/status-im/status-go/logutils"
+)
+
+// logAndCall logs request call details and executes the fn function if logging is enabled
+func logAndCall[T any](method string, params any, fn func() (T, error)) (T, error) {
+	var startTime time.Time
+	var duration time.Duration
+
+	if logutils.IsRequestLoggingEnabled() {
+		startTime = time.Now()
+	}
+
+	resp, err := fn()
+
+	if logutils.IsRequestLoggingEnabled() {
+		duration = time.Since(startTime)
+		logutils.GetRequestLogger().Debug(method, "params", fmt.Sprintf("%+v", params), "resp", fmt.Sprintf("%+v", resp), "err", err, "duration", duration)
+	}
+
+	return resp, err
+}
+
+func logAndCallString(method string, params any, fn func() string) string {
+	resp, _ := logAndCall(method, params, func() (string, error) {
+		return fn(), nil
+	})
+	return resp
+}
+
+func logAndCallVoid(method string, params any, fn func()) {
+	_, _ = logAndCall(method, params, func() (any, error) {
+		fn()
+		return nil, nil
+	})
+}
+
+func removeSensitiveInfo(jsonStr string) string {
+	// see related test for the usage of this function
+	re := regexp.MustCompile(`(?i)(".*?(password|mnemonic).*?")\s*:\s*("[^"]*")`)
```

Comment:
agian - only test builds, e2e / PR builds. I'll make sure it is tested with different kinds of builds

Reply:
Also, some local file paths could appear. Don't know if that's considered private.
https://github.com/status-im/status-go/blob/f660be0daa7cc742a07f6a139ea5ac9966f3ebe0/protocol/requests/create_account.go#L37

Reply:
I consider absolute paths private and I tend to remove them from the output of some tools whenever I can. For example from https://git-scm.com/docs/git-diagnose or from the Emacs bug report feature. But I don't know as well what's right legally speaking.

To me seems alright to log because we are talking about logging only in non-release builds and even in the worst case where there's a bug in our logic to log or not req/res per env, it would still be a small nuisance and we would identify and fix the bug in the build process.